plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -207,19 +215,71 @@ app_get_flatpak_ref_name (GsApp *app)
 	return gs_app_get_metadata_item (app, "flatpak::RefName");
 }
 
+static gchar **
+get_flatpak_default_locales (GsPlugin *plugin)
+{
+	g_auto(GStrv) flatpak_default_locales = NULL;
+	g_autoptr(GError) local_error = NULL;
+
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	if (priv->installation == NULL)
+		return NULL;
+
+	flatpak_default_locales = flatpak_installation_get_default_locales (priv->installation, &local_error);
+	if (local_error != NULL) {
+		g_warning ("No user locales could be retrieved! %s", local_error->message);
+		return NULL;
+	}
+
+	return g_steal_pointer (&flatpak_default_locales);
+}
+
+static gboolean
+assert_valid_locale_match (const char *app_locale,
+                           char **locale_options)
+{
+	int idx;
+	g_auto(GStrv) app_locale_parts = g_strsplit (app_locale, "_", -1);
```

Comment:
https://gitlab.gnome.org/GNOME/gnome-desktop/blob/master/libgnome-desktop/gnome-languages.h#L39-43 exists to parse locales.

Reply:
(And we [explicitly enable gnome-desktop support](https://github.com/endlessm/gnome-software/blob/921cae46b8656370dfdc21e97c3cffed3da4eaf4/debian/rules#L10), so gnome-software already has it as a dependency, although you might need to change the `meson.build` for the flatpak plugin to link to it.)